### ISSUE TITLE
JSON parser needs to account for `id`

### ISSUE THREAD

**Issue by gzt5142** [gzt5142/nldi-crawler-py#22]
In parsing the GeoJSON from source 12, I find that there's a bug in how we're processing features.  

The source table identifies the name of a column within `property` member that uniquely identifies each feature. All of the data we've got back from other sources do this.  Source 12 is strictly compliant with the most recent [GeoJSON format specification](https://www.rfc-editor.org/rfc/rfc7946#section-3.2), and returns the unique ID as a separate member: 

```
{
    "type": "Feature",
    "properties": {
        "description": "Well drilled or set into subsurface...",
        "name": "Water Well",
        "@iot.selfLink": "https://st2.newmexicowaterdata.org/FROST-Server/v1.1/Things(3798)",
        "Locations": [ ... ] , 
        "Datastreams": [ ... ],
        "//": "Many other properties..."
        "WellID": "C08EDC8F-5365-4A0C-B472-F39F8785D52A",
        "agency": "NMBGMR",
        "PointID": "QY-0070",
        "WellDepth": 140.0,
        "source_id": 31305.0,
        "GeologicFormation": "231CHNL",
        "Altitude": 4130.0,
        "geoconnex": "https://geoconnex.us/nmwdi/st/locations/3798"
    },
    "geometry": {
        "type": "Point",
        "coordinates": [
            -103.6153979576192,
            35.071781644432996
        ]
    },
    "id": "3798"
}
```

Note that `id` is not a `property`.  It is a member of this feature at the same level of the JSON tree as `type`, `properties`, and `geometry`. Just like the **2016 GeoJSON** spec says it should be. 

In all the other sources, a unique ID could be found in the properties member. I suppose we could do that here also, with the `PointID` field. But a more compliant parser would recognize the `id` member (if present) and use it.

**Comment by gzt5142** [comment]
It's actually more complicated than that (go figure). 

From source 2 -- HUC12 pour points

```
{
    "type": "Feature",
    "id": "fpp.7",
    "geometry": {
        "type": "Point",
        "coordinates": [
            -70.63039745535151,
            41.760479055712324
        ]
    },
    "properties": {
        "FDCODE": 2,
        "FILL": 0,
        "PCTPOUR": 43.3405,
        "VPCTPOUR": 43.2936,
        "DOWNHUC": "",
        "HUC_12": "010900020303",
        "HU_12_DS": "OCEAN",
        "ISSINK": 0,
        "DOWNFLAG": 2,
        "TOHUC_DIST": -1
    }
}
```
While there is a GeoJSON2016-compliant `id` field, it's not nearly as useful as the `properties['HUC_12']` value we are currently using.

Thing is -- the unique key to this feature once it gets into the NLDI database is merely used as a unique key (in combination with the source ID number)... so it's not designed to be human readable.  The `id` member will work just fine (so long as we trust this data source -- but that was just as strong a caveat when we use the `HUC_12` property).  

The way forward seems to be to use the `id` member if present, else use a designated column from `properties` as the key value.

**Comment by gzt5142** [comment]
Easy fix:
 
```python
try:
  _my_id = itm["id"]
except KeyError:
  _my_id = itm["properties"][id]
```

**Comment by gzt5142** [comment]
It looks like more sources than I had originally appreciated are using the GeoJSON2016-cmpliant format. Only a couple of sources **don't** use it.

**Comment by gzt5142** [comment]
closed by #24

**Comment by dblodgett-usgs** [comment]
Good catch on this one -- I think the java crawler accounted for it in its own way. The variable use of the `id` in geojson has always bugged me.
